packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -937,12 +937,21 @@ public void moveCamera(@NonNull Messages.PlatformCameraUpdate cameraUpdate) {
   }
 
   @Override
-  public void animateCamera(@NonNull Messages.PlatformCameraUpdate cameraUpdate) {
+  public void animateCamera(
+      @NonNull Messages.PlatformCameraUpdate cameraUpdate,
+      @Nullable Messages.PlatformCameraUpdateAnimationConfiguration configuration) {
     if (googleMap == null) {
       throw new FlutterError(
           "GoogleMap uninitialized", "animateCamera called prior to map initialization", null);
     }
-    googleMap.animateCamera(Convert.cameraUpdateFromPigeon(cameraUpdate, density));
+    if (configuration != null && configuration.getDurationMilliseconds() != null) {
+      googleMap.animateCamera(
+          Convert.cameraUpdateFromPigeon(cameraUpdate, density),
+          configuration.getDurationMilliseconds().intValue(),
+          null);
+    } else {
+      googleMap.animateCamera(Convert.cameraUpdateFromPigeon(cameraUpdate, density));
+    }
```

Comment:
There should be a native unit test validating the branching here (via a mock map).